### ISSUE TITLE
[expo-image] Attempted import error: 'ExpoImageModule' is not exported from './ExpoImage'. 

### ISSUE THREAD

**Issue by sergeu90** [expo/expo#21846]
### Summary

Run on the web with expo-image package.
![image](https://user-images.githubusercontent.com/7217517/227792023-0fffdb5e-721f-4b50-b154-9326ea3a52be.png)

### What platform(s) does this occur on?

Web

### SDK Version

48.0.9

### Environment

expo-env-info 1.0.5 environment info:
System:
OS: macOS 13.0.1
Shell: 5.8.1 - /bin/zsh
Binaries:
Node: 14.21.3 - /usr/local/bin/node
Yarn: 1.22.19 - /usr/local/bin/yarn
npm: 6.14.18 - /usr/local/bin/npm
Managers:
CocoaPods: 1.11.2 - /usr/local/bin/pod
SDKs:
iOS SDK:
Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
Android SDK:
API Levels: 32, 33
Build Tools: 32.0.0, 32.1.0, 33.0.0, 33.0.0
System Images: android-25 | Google APIs ARM 64 v8a, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-30 | Google APIs ATD ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-32 | Google Play ARM 64 v8a, android-33 | Google Play ARM 64 v8a
IDEs:
Android Studio: 2021.2 AI-212.5712.43.2112.8609683
Xcode: 14.2/14C18 - /usr/bin/xcodebuild
npmPackages:
@expo/webpack-config: 0.17.4 => 0.17.4
expo: 48.0.9 => 48.0.9
react: 18.2.0 => 18.2.0
react-dom: 18.2.0 => 18.2.0
react-native: 0.71.4 => 0.71.4
react-native-web: 0.18.12 => 0.18.12
npmGlobalPackages:
eas-cli: 3.8.1
expo-cli: 6.1.0
Expo Workflow: managed

### Minimal reproducible example

```
import { Image } from 'expo-image';

const Test = () => {
  return (
      <Image
        style={styles.logo}
        source={ImageUrl}
        resizeMode="contain"
       />
};
```

**Comment by lrpinto** [comment]
I have the same issue. Attempting to run with Web Development Server.

Does not stops app from running, but Image does not display, and terminal shows following output.
```
WARNING in ./node_modules/expo-image/build/Image.js:36:17
export 'ExpoImageModule' (imported as 'ExpoImageModule') was not found in './ExpoImage' (possible exports: default)
  34 |    */
  35 |   static async clearDiskCache(): Promise<boolean> {
> 36 |     return await ExpoImageModule.clearDiskCache();
     |                 ^
  37 |   }
  38 |
  39 |   render() {
```
  
  My Package.json
  
`{
  "name": "app-version-2",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "@expo/webpack-config": "^18.0.1",
    "@types/react": "~18.0.27",
    "expo": "~48.0.6",
    "expo-image": "~1.0.0",
    "expo-status-bar": "~1.4.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.71.4",
    "react-native-web": "~0.18.10",
    "typescript": "^4.9.4"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@storybook/addon-actions": "^6.5.16",
    "@storybook/addon-essentials": "^6.5.16",
    "@storybook/addon-interactions": "^6.5.16",
    "@storybook/addon-links": "^6.5.16",
    "@storybook/react": "^6.5.16",
    "@storybook/testing-library": "^0.0.13",
    "babel-loader": "^8.3.0"
  },
  "private": true
}
`

**Comment by sergeu90** [comment]
@lrpinto I see image and all work good but in the console i see error about methods this. If don't use this methods on the web all will works but i think need fix for the web version.

**Comment by Crafter-Y** [comment]
I'm getting the same Error. 

```
[2] WARNING in ./node_modules/expo-image/build/Image.js:18:11
[2] export 'ExpoImageModule' (imported as 'ExpoImageModule') was not found in './ExpoImage' (possible exports: default)
[2]   16 |    */
[2]   17 |   static prefetch(urls: string | string[]): void {
[2] > 18 |     return ExpoImageModule.prefetch(Array.isArray(urls) ? urls : [urls]);
[2]      |           ^
[2]   19 |   }
[2]   20 | 
[2]   21 |   /**
[2]
[2] WARNING in ./node_modules/expo-image/build/Image.js:27:17
[2] export 'ExpoImageModule' (imported as 'ExpoImageModule') was not found in './ExpoImage' (possible exports: default)
[2]   25 |    */
[2]   26 |   static async clearMemoryCache(): Promise<boolean> {
[2] > 27 |     return await ExpoImageModule.clearMemoryCache();
[2]      |                 ^
[2]   28 |   }
[2]   29 | 
[2]   30 |   /**
[2]
[2] WARNING in ./node_modules/expo-image/build/Image.js:36:17
[2] export 'ExpoImageModule' (imported as 'ExpoImageModule') was not found in './ExpoImage' (possible exports: default)
[2]   34 |    */
[2]   35 |   static async clearDiskCache(): Promise<boolean> {
[2] > 36 |     return await ExpoImageModule.clearDiskCache();
[2]      |                 ^
[2]   37 |   }
[2]   38 | 
[2]   39 |   render() {
```

Im my case, this appears when starting the web `w`

**Comment by 0x10Ayoub** [comment]
using require in the source code seems to fix the issues 
const {ExpoImageModule} = require('./ExpoImage') in the ****/image.js
anyone know why it does work and ECS import doesn't work

**Comment by izakfilmalter** [comment]
Any progress on this? This is holding us back from upgrading next. I think the fix is as simple as having a web version of `Image.tsx`:
```
import React from 'react';
import { StyleSheet } from 'react-native';

import ExpoImage from './ExpoImage.web';
import { ImageProps } from './Image.types';
import { resolveContentFit, resolveContentPosition, resolveTransition } from './utils';
import { resolveSources } from './utils/resolveSources';

let loggedDefaultSourceDeprecationWarning = false;

export class Image extends React.PureComponent<ImageProps> {
  render() {
    const {
      style,
      source,
      placeholder,
      contentFit,
      contentPosition,
      transition,
      fadeDuration,
      resizeMode: resizeModeProp,
      defaultSource,
      loadingIndicatorSource,
      ...restProps
    } = this.props;

    const { resizeMode: resizeModeStyle, ...restStyle } = StyleSheet.flatten(style) || {};
    const resizeMode = resizeModeProp ?? resizeModeStyle;

    if ((defaultSource || loadingIndicatorSource) && !loggedDefaultSourceDeprecationWarning) {
      console.warn(
        '[expo-image]: `defaultSource` and `loadingIndicatorSource` props are deprecated, use `placeholder` instead'
      );
      loggedDefaultSourceDeprecationWarning = true;
    }

    return (
      <ExpoImage
        {...restProps}
        style={restStyle}
        source={resolveSources(source)}
        placeholder={resolveSources(placeholder ?? defaultSource ?? loadingIndicatorSource)}
        contentFit={resolveContentFit(contentFit, resizeMode)}
        contentPosition={resolveContentPosition(contentPosition)}
        transition={resolveTransition(transition, fadeDuration)}
      />
    );
  }
}

```

**Comment by aleqsio** [comment]
Hey folks!

I would appreciate a repository reproduction – if you can reproduce this please share the whole code sample :)
For me this works on web:
```tsx
import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import { Image } from "expo-image";

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Open up App.js to start working on your app!</Text>
      <Image
        style={styles.image}
        source={{ uri: "https://picsum.photos/200" }}
        resizeMode="contain"
      />
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
  },
  image: {
    flex: 1,
    backgroundColor: "red",
  },
});

```
<img width="300" alt="image" src="https://github.com/expo/expo/assets/5597580/648e500f-7bac-4ac7-ac21-bf992ed4d710">

**Comment by aleqsio** [comment]
I'm going to keep this open for now, want to figure out what's broken for you

**Comment by llllvvuu** [comment]
This is reproducible in https://github.com/bluesky-social/social-app/tree/15c1b6ee157471807a723161066ba4ce5e12c0b5 (I realize this is far from minimal though)

**Comment by SleeplessByte** [comment]
@aleqsio I'll create a minimal repro for you. Saw this immediately after adding `expo-image` to an SDK 48 project.

The issue is that:

[ExpoImage.tsx](https://github.com/expo/expo/blob/main/packages/expo-image/src/ExpoImage.tsx#LL118C10-L118C26)

...exports `ExpoImageModule` as a named export

And:

[ExpoImage.web.tsx](https://github.com/expo/expo/blob/main/packages/expo-image/src/ExpoImage.web.tsx) does not.

**Comment by SleeplessByte** [comment]
Here is the minimal repro from a fresh expo project using `create-expo-app`.

https://github.com/XPBytes/issue-expo-image-import

**Comment by aleqsio** [comment]
@SleeplessByte The repro shows an image once you set `style={{ width: 200, height: 200 }}` on the component.

At the same time, will fix the import warnings but these functions are not yet implemented on web so these functions will be a noop.

**Comment by SleeplessByte** [comment]
My repo was for the warnings :)

Thanks for picking it up!

**Comment by aleqsio** [comment]
Yeah, I get to fix two issues at the price of one, the width, height would also make sense to be the default size of the image if no styles are set :)

**Comment by aleqsio** [comment]
This will be fixed in new expo-image version

**Comment by RichardLindhout** [comment]
@aleqsio when will this be released?

**Comment by buster95** [comment]
I realize if you use metro bundler problem dissapear and image render
`app.json`
```json
{
   "expo": {
      "web": {
         "bundler": "metro"
      }
   }
}
```

but I lost hot reload

**Comment by aleqsio** [comment]
Sorry for missing this thread, it was published in  (1.3.1) as `next` initially and we've promoted it to latest recently.
